src/werkzeug/local.py

```diff
@@ -46,41 +89,59 @@ def release_local(local: Union["LocalStack", "Local"]) -> None:
 
 
 class Local:
-    __slots__ = ("__storage__", "__ident_func__")
+    __slots__ = ("_storage", "_ident_func")
 
     def __init__(self) -> None:
-        object.__setattr__(self, "__storage__", {})
-        object.__setattr__(self, "__ident_func__", get_ident)
+        object.__setattr__(self, "_storage", ContextVar("local_storage"))
+        object.__setattr__(self, "_ident_func", get_ident)
+
+    @property
+    def __storage__(self):
+        return self._storage.get({})
+
+    @property
+    def __ident_func__(self):
+        return self._ident_func
```

Comment:
This should match the setter, it doesn't seem like there's a point on storing `_ident_func` here anymore.

Should also show a deprecation warning as it will not be used once 3.6 is dropped.

Reply:
Done